servers/rendering/renderer_rd/renderer_scene_sky_rd.cpp

```diff
@@ -0,0 +1,1345 @@
+		RD::get_singleton()->free(radiance);
+		radiance = RID();
+	}
+	reflection.clear_reflection_data();
+
+	return true;
+}
+
+bool RendererSceneSkyRD::Sky::set_mode(RS::SkyMode p_mode) {
+	if (mode == p_mode) {
+		return false;
+	}
+
+	mode = p_mode;
+
+	if (mode == RS::SKY_MODE_REALTIME && radiance_size != 256) {
+		WARN_PRINT("Realtime Skies can only use a radiance size of 256. Radiance size will be set to 256 internally.");
+		set_radiance_size(256);
+	}
+
+	if (radiance.is_valid()) {
+		RD::get_singleton()->free(radiance);
+		radiance = RID();
+	}
+	reflection.clear_reflection_data();
+
+	return true;
+}
+
+bool RendererSceneSkyRD::Sky::set_material(RID p_material) {
+	if (material == p_material) {
+		return false;
+	}
+
+	material = p_material;
+	return true;
+}
+
+Ref<Image> RendererSceneSkyRD::Sky::bake_panorama(RendererStorageRD *p_storage, float p_energy, int p_roughness_layers, const Size2i &p_size) {
+	if (radiance.is_valid()) {
+		RD::TextureFormat tf;
+		tf.format = RD::DATA_FORMAT_R32G32B32A32_SFLOAT;
+		tf.width = p_size.width;
+		tf.height = p_size.height;
+		tf.usage_bits = RD::TEXTURE_USAGE_STORAGE_BIT | RD::TEXTURE_USAGE_CAN_COPY_FROM_BIT;
+
+		RID rad_tex = RD::get_singleton()->texture_create(tf, RD::TextureView());
+		p_storage->get_effects()->copy_cubemap_to_panorama(radiance, rad_tex, p_size, p_roughness_layers, reflection.layers.size() > 1);
+		Vector<uint8_t> data = RD::get_singleton()->texture_get_data(rad_tex, 0);
+		RD::get_singleton()->free(rad_tex);
+
+		Ref<Image> img;
+		img.instance();
+		img->create(p_size.width, p_size.height, false, Image::FORMAT_RGBAF, data);
+		for (int i = 0; i < p_size.width; i++) {
+			for (int j = 0; j < p_size.height; j++) {
+				Color c = img->get_pixel(i, j);
+				c.r *= p_energy;
+				c.g *= p_energy;
+				c.b *= p_energy;
+				img->set_pixel(i, j, c);
+			}
+		}
+		return img;
+	}
+
+	return Ref<Image>();
+}
+
+RID RendererSceneSkyRD::Sky::get_radiance_uniform_set_rd(RID p_shader, int p_set) {
+	if (uniform_set.is_null() || !RD::get_singleton()->uniform_set_is_valid(uniform_set)) {
+		uniform_set = RID();
+		if (radiance.is_valid()) {
+			Vector<RD::Uniform> uniforms;
+			{
+				RD::Uniform u;
+				u.uniform_type = RD::UNIFORM_TYPE_TEXTURE;
+				u.binding = 0;
+				u.ids.push_back(radiance);
+				uniforms.push_back(u);
+			}
+
+			uniform_set = RD::get_singleton()->uniform_set_create(uniforms, p_shader, p_set);
+		}
+	}
+
+	return uniform_set;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// RendererSceneSkyRD
+
+RendererStorageRD::ShaderData *RendererSceneSkyRD::_create_sky_shader_func() {
+	SkyShaderData *shader_data = memnew(SkyShaderData);
+	return shader_data;
+}
+
+RendererStorageRD::ShaderData *RendererSceneSkyRD::_create_sky_shader_funcs() {
+	// !BAS! Why isn't _create_sky_shader_func not just static too?
+	return static_cast<RendererSceneRenderRD *>(RendererSceneRenderRD::singleton)->sky._create_sky_shader_func();
```

Comment:
@reduz this was just a reminder for me to ask you, I noticed this when I worked on this before clayjohn. Why are these functions split in this way?